presto-native-execution/src/test/java/com/facebook/presto/nativeworker/TestHiveQueriesGlueMetastore.java

```diff
@@ -0,0 +1,446 @@
+    private String schemaName;
+
+    private final boolean useThrift;
+
+    protected TestHiveQueriesGlueMetastore(boolean useThrift)
+    {
+        super(useThrift);
+        this.useThrift = useThrift;
+    }
+    @Override
+    protected QueryRunner createQueryRunner() throws Exception
+    {
+        String prestoServerPath = System.getProperty("PRESTO_SERVER");
+        String dataDirectory = System.getProperty("DATA_DIR");
+        String workerCount = System.getProperty("WORKER_COUNT");
+        int cacheMaxSize = 0;
+
+        assertNotNull(prestoServerPath, "Native worker binary path is missing. Add -DPRESTO_SERVER=<path/to/presto_server> to your JVM arguments.");
+        assertNotNull(dataDirectory, "Data directory path is missing. Add -DDATA_DIR=<path/to/data> to your JVM arguments.");
+
+        return HiveExternalWorkerQueryRunner.createNativeQueryRunner(
+                dataDirectory,
+                prestoServerPath,
+                Optional.ofNullable(workerCount).map(Integer::parseInt),
+                cacheMaxSize,
+                useThrift);
+    }
+
+    @Override
+    protected ExpectedQueryRunner createExpectedQueryRunner() throws Exception
+    {
+        String dataDirectory = System.getProperty("DATA_DIR");
+        return HiveExternalWorkerQueryRunner.createJavaQueryRunner(Optional.of(Paths.get(dataDirectory)));
+    }
+
+    private static void setUpGlueCatalog(QueryRunner queryRunner, File tempWarehouseDir) throws IOException
+    {
+        assertNotNull(queryRunner, "Query runner must be provided");
+        assertNotNull(tempWarehouseDir, "Temp warehouse directory must be provided");
+
+        queryRunner.installPlugin(new HivePlugin("glue"));
+        Map<String, String> glueProperties = ImmutableMap.<String, String>builder()
+                .put("hive.metastore", "glue")
+                .put("hive.metastore.glue.region", "us-east-1")
+                .put("hive.metastore.glue.default-warehouse-dir", tempWarehouseDir.getCanonicalPath())
+                // This option is required to avoid the error:
+                // "Table scan with filter pushdown disabled is not supported".
+                // Velox requires this flag to be set for scans.
+                .put("hive.pushdown-filter-enabled", "true")
+                // The options below are required to avoid "Access Denied" errors when performing operations such
+                // as DROP TABLE.
+                .put("hive.max-partitions-per-writers", "999")
+                .put("hive.allow-drop-table", "true")
+                .put("hive.allow-rename-table", "true")
+                .put("hive.allow-rename-column", "true")
+                .put("hive.allow-add-column", "true")
+                .put("hive.allow-drop-column", "true")
+                .build();
+
+        queryRunner.createCatalog("test_glue", "glue", glueProperties);
+    }
+
+    @Override
+    protected Session getSession()
```

Comment:
Can you please reorder the methods with the access level descending? public, protected, default, private

Reply:
I would prefer to keep them in this logical order.

Reply:
I updated the order, I just left protected constructor at the top of the file.